### PR TITLE
mobilebert model card update

### PR DESCRIPTION
**Pull request by Reshan123**
# What does this PR do?
As suggested in this issue - [Model cards #36979](https://github.com/huggingface/transformers/issues/36979) - this PR updates the documentation of the MobileBERT model, which will now be aligned with the standardized format for all the docs.

## Who can review?
@stevhliu, please let me know if any changes are needed

### CODE REVIEW

File: docs/source/en/model_doc/mobilebert.md [hunk 1]
```diff
@@ -14,43 +14,86 @@ rendered properly in your Markdown viewer.
...
+        <img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-FF6F00?style=flat&logo=tensorflow&logoColor=white">
+    </div>
 </div>
 
-## Overview
+# MobileBERT
 
-The MobileBERT model was proposed in [MobileBERT: a Compact Task-Agnostic BERT for Resource-Limited Devices](https://arxiv.org/abs/2004.02984) by Zhiqing Sun, Hongkun Yu, Xiaodan Song, Renjie Liu, Yiming Yang, and Denny
-Zhou. It's a bidirectional transformer based on the BERT model, which is compressed and accelerated using several
+[MobileBERT](https://github.com/google-research/google-research/tree/master/mobilebert) is a lightweight and efficient variant of BERT, specifically designed for resource-limited devices such as mobile phones. It retains the bidirectional transformer architecture of BERT but significantly reduces model size and inference latency while maintaining strong performance on NLP tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilebert.md:27]:
```suggestion
[MobileBERT](https://huggingface.co/papers/2004.02984) is a lightweight and efficient variant of BERT, specifically designed for resource-limited devices such as mobile phones. It retains BERT's architecture but significantly reduces model size and inference latency while maintaining strong performance on NLP tasks. MobileBERT achieves this through a bottleneck structure and carefully balanced self-attention and feedforward networks. The model is trained by knowledge transfer from a large BERT model with an inverted bottleneck structure.
```

File: docs/source/en/model_doc/mobilebert.md [hunk 2]
```diff
@@ -14,43 +14,86 @@ rendered properly in your Markdown viewer.
...
+The MobileBERT model was proposed in [MobileBERT: a Compact Task-Agnostic BERT for Resource-Limited Devices](https://arxiv.org/abs/2004.02984). It's a bidirectional transformer based on the BERT model, which is compressed and accelerated using several
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilebert.md:28]: We can remove this paragraph

File: docs/source/en/model_doc/mobilebert.md [hunk 3]
```diff
@@ -14,43 +14,86 @@ rendered properly in your Markdown viewer.
...
 approaches.
 
-The abstract from the paper is the following:
+> [!TIP]
+> Click on the  models in the right sidebar for more examples of how to apply Gemma to different vision and language tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilebert.md:32]:
```suggestion
You can find the original MobileBERT checkpoint under the [Google](https://huggingface.co/google/mobilebert-uncased) organization.

> [!TIP]
> Click on the MobileBERT models in the right sidebar for more examples of how to apply MobileBERT to different language tasks.
```

File: docs/source/en/model_doc/mobilebert.md [hunk 4]
```diff
@@ -14,43 +14,86 @@ rendered properly in your Markdown viewer.
...
-the popular BERT model. Like the original BERT, MobileBERT is task-agnostic, that is, it can be generically applied to
-various downstream NLP tasks via simple fine-tuning. Basically, MobileBERT is a thin version of BERT_LARGE, while
-equipped with bottleneck structures and a carefully designed balance between self-attentions and feed-forward networks.
-To train MobileBERT, we first train a specially designed teacher model, an inverted-bottleneck incorporated BERT_LARGE
-model. Then, we conduct knowledge transfer from this teacher to MobileBERT. Empirical studies show that MobileBERT is
-4.3x smaller and 5.5x faster than BERT_BASE while achieving competitive results on well-known benchmarks. On the
-natural language inference tasks of GLUE, MobileBERT achieves a GLUEscore o 77.7 (0.6 lower than BERT_BASE), and 62 ms
-latency on a Pixel 4 phone. On the SQuAD v1.1/v2.0 question answering task, MobileBERT achieves a dev F1 score of
-90.0/79.2 (1.5/2.1 higher than BERT_BASE).*
+## Usage tips
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilebert.md:34]: We can remove the entire `## Usage tips` section here

File: docs/source/en/model_doc/mobilebert.md [hunk 5]
```diff
@@ -14,43 +14,86 @@ rendered properly in your Markdown viewer.
...
+    * SQuAD v1.1/v2.0: F1 scores of 90.0/79.2 (outperforming BERT_BASE by 1.5/2.1 points).
 
-## Usage tips
 
-- MobileBERT is a model with absolute position embeddings so it's usually advised to pad the inputs on the right rather
-  than the left.
-- MobileBERT is similar to BERT and therefore relies on the masked language modeling (MLM) objective. It is therefore
-  efficient at predicting masked tokens and at NLU in general, but is not optimal for text generation. Models trained
-  with a causal language modeling (CLM) objective are better in that regard.
+The example below demonstrates how to generate text with [`Pipeline`] or the [`AutoModel`], and from the command line.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilebert.md:44]:
```suggestion
The example below demonstrates how to predict the `[MASK]` token with [`Pipeline`], [`AutoModel`], and from the command line.
```

File: docs/source/en/model_doc/mobilebert.md [hunk 6]
```diff
@@ -14,43 +14,86 @@ rendered properly in your Markdown viewer.
...
+
+<hfoptions id="usage">
+<hfoption id="Pipeline">
+
+```py
+from transformers import pipeline
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilebert.md:50]: Snippet should look like below. You can remove the code comments too

```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="fill-mask",
    model="google/mobilebert-uncased",
    torch_dtype=torch.float16,
    device=0
)
pipeline("The capital of France is [MASK].")
```

File: docs/source/en/model_doc/mobilebert.md [hunk 7]
```diff
@@ -14,43 +14,86 @@ rendered properly in your Markdown viewer.
...
+
+# Example: Predict the masked word
+results = mask_filler("The capital of France is [MASK].")
+print(results[0]["sequence"]) # Output: "The capital of France is paris."
+```
+</hfoption>
+<hfoption id="AutoModel">
+
+```py
+from transformers import AutoTokenizer, AutoModelForMaskedLM
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilebert.md:63]: The code snippet should look like below. You can also remove the code comments here!

```py
import torch
from transformers import AutoModelForMaskedLM, AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained(
    "google/mobilebert-uncased",
)
model = AutoModelForMaskedLM.from_pretrained(
    "google/mobilebert-uncased",
    torch_dtype=torch.float16,
    device_map="auto",
)
inputs = tokenizer("The capital of France is [MASK].", return_tensors="pt").to("cuda")

with torch.no_grad():
    outputs = model(**inputs)
    predictions = outputs.logits

masked_index = torch.where(inputs['input_ids'] == tokenizer.mask_token_id)[1]
predicted_token_id = predictions[0, masked_index].argmax(dim=-1)
predicted_token = tokenizer.decode(predicted_token_id)

print(f"The predicted token is: {predicted_token}")
```

File: docs/source/en/model_doc/mobilebert.md [hunk 8]
```diff
@@ -14,43 +14,86 @@ rendered properly in your Markdown viewer.
...
+# Decode the prediction
+predicted_token = tokenizer.decode(predicted_token_id)
+print(f"Predicted word: {predicted_token}")  # Output: "buy" or "order"
+```
+
+</hfoption>
+<hfoption id="transformers-cli">
+
+```bash
+python -c "from transformers import pipeline; print(pipeline('fill-mask', model='google/mobilebert-uncased')('Artificial intelligence will [MASK] the world.')[0]['sequence'])"
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilebert.md:92]:
```suggestion
echo -e "The capital of France is [MASK]." | transformers-cli run --task fill-mask --model google/mobilebert-uncased --device 0
```

File: docs/source/en/model_doc/mobilebert.md [hunk 9]
```diff
@@ -14,43 +14,86 @@ rendered properly in your Markdown viewer.
...
+```
+
+</hfoption>
+</hfoptions>
 
 
 ## Resources
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilebert.md:99]: Remove `## Resources` and replace it with `## Notes`:

```md
## Notes

- Inputs should be padded on the right because BERT uses absolute position embeddings.
```

**Review by stevhliu — COMMENTED** [review]
Thanks for your contribution!

**Review by stevhliu — APPROVED** [review]
Thanks for your contribution! 🤗

**Review by stevhliu — APPROVED** [review]
👍
